src/opendlv-statsd.cpp

```diff
@@ -47,27 +59,90 @@ int32_t main(int32_t argc, char **argv) {
         //       would have to be created and registered.
 
         // Handler to receive a GPS message (realized as C++ lambda).
-        auto onGPS = [&dataMutex](cluon::data::Envelope &&env){
+        
+        std::vector<double> lat_queue,lon_queue; //vector is used as a queue
+        
+        auto onGPS = [&dataMutex,&lat_queue,&lon_queue,VERBOSE](cluon::data::Envelope &&env){
             // Now, we unpack the cluon::data::Envelope to get the desired WGS84Reading.
             opendlv::proxy::GeodeticWgs84Reading gps = cluon::extractMessage<opendlv::proxy::GeodeticWgs84Reading>(std::move(env));
 
             {
+                //retrieve latitude and longitude from the GeodeticWgs84Reading
+                double latitude=gps.latitude();
+                double longitude=gps.longitude();   
+                
+                if(VERBOSE) 
+                    std::cerr << "Got " << latitude << ", " << longitude << " as Envelopes" <<std::endl;
+                
+                std::lock_guard<std::mutex> lck(dataMutex);
+                
+                //push the data into queues
+                lat_queue.insert(lat_queue.end(),latitude);
+                lon_queue.insert(lon_queue.end(),longitude);
+
                 // TODO: Temporarily store the desired field to be passed on to statsd in the time-trigger below.
             }
         };
-
+        
+        
         // Register our lambda for the message identifier for opendlv::proxy::GeodeticWgs84Reading.
         od4.dataTrigger(opendlv::proxy::GeodeticWgs84Reading::ID(), onGPS);
-
+        
         // Time-triggered sender; TODO: reflect the desired sample rate via commandline.
-        const float ONE_HZ{1.0f};
-        od4.timeTrigger(ONE_HZ, [&od4, &dataMutex](){
+        float FREQUENCY{std::stof(sample_rate+"f")};
```

Comment:
The frequency for `timeTrigger` is expecting the value in `Hz`; so, you can update your semantics at the command line to avoid this std::stof(..) parsing here.